test/gurka/test_matrix.cc

```diff
@@ -538,3 +538,123 @@ TEST(StandAlone, CostMatrixShapes) {
 
   res.erase();
 }
+
+class DateTimeTest : public ::testing::Test {
+protected:
+  // check both with and without time zones present
+  static gurka::map map;
+  static gurka::map map_tz;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 1500;
+
+    // ~ are approximate time zone crossings
+    const std::string ascii_map = R"(
+      A----------B
+      |          |
+      C          D
+      |          |
+      ~          ~
+      |          |
+      |          |
+      E          F
+      |          |
+      G----------H
+    )";
```

Comment:
this is really good too! not sure we had any end-to-end test with timezones derived from a gurka map all the way to checking the date_time in the response